mdonatello/mdonatello.py

```diff
@@ -116,78 +159,229 @@ def draw_molecule(self, mol, show_atom_indices, width, height):
+            fg_counts = self.calculate_functional_groups(mol)
+            fg_checkboxes = []
+            for fg, atom_indices in fg_counts.items():
+                if atom_indices:
+                    fg_checkbox_name = f"fg_checkbox_{fg}"
+                    if not hasattr(self, fg_checkbox_name):
+                        checkbox = Checkbox(value=False, description=fg)
+                        setattr(self, fg_checkbox_name, checkbox)
+                        checkbox.observe(self.update_display, names="value")
+                    fg_checkboxes.append(getattr(self, fg_checkbox_name))
+            
+            if fg_checkboxes:
+                fg_hbox = HBox(fg_checkboxes)
+                children.append(functional_groups_header)
+                children.append(fg_hbox)
         
         self.output_molecule.children = children
         
     def display_molecular_weight(self, mol):
         mw = Descriptors.MolWt(mol)
-        return HTML(f"<p style='margin: 0; margin-left: 100px;'>Molecular Weight: {mw:.2f} g/mol</p>")
+        return HTML(
+            f"<p style='margin: 0; margin-left: 100px;'>Molecular Weight: {mw:.2f} g/mol</p>"
+        )
         
     def display_logp(self, mol):
         logp = Descriptors.MolLogP(mol)
-        return HTML(f"<p style='margin: 0; margin-left: 100px;'>LogP: {logp:.2f}</p>")
+        return HTML(
+            f"<p style='margin: 0; margin-left: 100px;'>LogP: {logp:.2f}</p>"
+        )
 
     def display_num_h_donors(self, mol):
         num_h_donors = Descriptors.NumHDonors(mol)
-        return HTML(f"<p style='margin: 0; margin-left: 100px;'>Number of H-Bond Donors: {num_h_donors}</p>")
+        return HTML(
+            f"<p style='margin: 0; margin-left: 100px;'>Number of H-Bond Donors: {num_h_donors}</p>"
+        )
 
     def display_num_h_acceptors(self, mol):
         num_h_acceptors = Descriptors.NumHAcceptors(mol)
-        return HTML(f"<p style='margin: 0; margin-left: 100px;'>Number of H-Bond Acceptors: {num_h_acceptors}</p>")
-    
+        return HTML(
+            f"<p style='margin: 0; margin-left: 100px;'>Number of H-Bond Acceptors: {num_h_acceptors}</p>"
+        )
+
     def display_tpsa(self, mol):
         tpsa = Descriptors.TPSA(mol)
-        return HTML(f"<p style='margin: 0; margin-left: 100px;'>Topological Polar Surface Area (TPSA): {tpsa:.2f} Å²</p>")
-        
+        return HTML(
+            f"<p style='margin: 0; margin-left: 100px;'>Topological Polar Surface Area (TPSA): {tpsa:.2f} Å²</p>"
+        )
+
     def display_rotatable_bonds(self, mol):
         rotatable_bonds = Descriptors.NumRotatableBonds(mol)
-        return HTML(f"<p style='margin: 0; margin-left: 100px;'>Number of Rotatable Bonds: {rotatable_bonds}</p>")
+        return HTML(
+            f"<p style='margin: 0; margin-left: 100px;'>Number of Rotatable Bonds: {rotatable_bonds}</p>"
+        )
+ 
+    def calculate_functional_groups(self, mol):
+        functional_groups = {
```

Comment:
there's a dictionary being created here, but it's never being used as a dictionary because you're only accessing it via `.items()`.  instead you could just create a list of tuples of (fg_name, fg_SMARTS).  so:
```python
functional_groups = [
    ('Hydroxyl_group (-OH)', '[OX2H]'),
    ...
]
```

Similarly I think the output of this function is a dict but never used as a dict but is instead immediately traversed via `.items()`.  If you're immediately consuming the output of this function it could even be a generator...